dedalus/tools/array.py

```diff
@@ -427,7 +429,19 @@ def matvec(x):
     evals, evecs = spla.eigs(D, k=N, which='LM', sigma=None, **kw)
     # Rectify eigenvalues
     evals = 1 / evals + target
-    return evals, evecs
+    if left:
+        # Build sparse linear operator representing (A^H - conj(σ)B^H)^I B^H = C^-H B^H = D_adj
+        # Note: D_adj is not equal to D^H
+        def matvec_adj(x):
+            return solver.solve_adjoint(B.H.dot(x))
+        D_adj = spla.LinearOperator(dtype=A.dtype, shape=A.shape, matvec=matvec_adj)
+        # Solve using scipy sparse algorithm
+        evals_adjoint, evecs_adjoint = spla.eigs(D_adj, k=N, which='LM', sigma=None, **kw)
```

Comment:
I think we should call these `left_` instead of `_adjoint` throughout to be consistent with scipy.

Reply:
Agreed, have updated the code.